pkg/operator/apiservicecontroller/apiservice_controller.go

```diff
@@ -230,3 +248,107 @@ func (c *APIServiceController) eventHandler() cache.ResourceEventHandler {
 		DeleteFunc: func(obj interface{}) { c.queue.Add(workQueueKey) },
 	}
 }
+
+// externalOperatorPreconditionFunc represents a function that will be called by this operator
+// to determine if an external operator is ready to take over some API services
+type externalOperatorPreconditionFuncType func() error
+
+// GetAPIServicesToManageWrapper returns a function that returns the desired list of API Services that will be managed by this operator
+// note that some services might be managed by an external operators/servers
+func GetAPIServicesToManageWrapper(apiregistrationv1Client apiregistrationv1client.ApiregistrationV1Interface,
+	operatorPrecondition externalOperatorPreconditionFuncType,
+	authoritativeAPIServicesList []*apiregistrationv1.APIService,
+	eventRecorder events.Recorder,
+	apiGroupsManagedByExternalServer sets.String,
+	apiGroupsManagedByExternalServerAnnotation string) func() ([]*apiregistrationv1.APIService, error) {
+
+	observedAPIServices := authoritativeAPIServicesList
+	return func() ([]*apiregistrationv1.APIService, error) {
+		if externalOperatorPrecondition := operatorPrecondition(); externalOperatorPrecondition != nil {
+			return nil, externalOperatorPrecondition
+		}
+
+		apiServicesToManage := []*apiregistrationv1.APIService{}
+		for _, apiService := range authoritativeAPIServicesList {
+			if apiGroupsManagedByExternalServer.Has(apiService.Name) && isAPIServiceAnnotatedByExternalServer(apiregistrationv1Client, apiService, apiGroupsManagedByExternalServerAnnotation) {
+				continue
+			}
+			apiServicesToManage = append(apiServicesToManage, apiService)
+		}
+
+		{
+			added, removed := diffAPIServices(observedAPIServices, apiServicesToManage)
+			if len(added) > 0 {
```

Comment:
if any changed, just write the entire new list.  I don't need separate message, but I only want messages for 

```
if len(added) >0 || len(removed) > 0{
   event("new list is %", list)
```